src/QPConstr.cpp

```diff
@@ -105,6 +124,8 @@ void JointLimitsConstr::update(const std::vector<rbd::MultiBody> & /* mbs */,
 
   lower_ = lower_.cwiseMax(alphaDLower_);
   upper_ = upper_.cwiseMin(alphaDUpper_);
+  lower_ = lower_.cwiseMax((alphaDDLower_ * step_) + curAlphaD_);
+  upper_ = upper_.cwiseMin((alphaDDUpper_ * step_) + curAlphaD_);
```

Comment:
`curAlphaD_` needs to be updated (and I would rename it `prevAlphaD_` I think this is clearer)